client/lib/after-layout-flush/index.js

```diff
@@ -18,7 +18,14 @@ export default function afterLayoutFlush( func ) {
 	let rafHandle = undefined;
 	let timeoutHandle = undefined;
 
+	const hasRAF = typeof requestAnimationFrame === 'function';
+
+	let lastThis, lastArgs;
+
 	const scheduleRAF = rafFunc => ( ...args ) => {
+		lastThis = this;
+		lastArgs = args;
```

Comment:
Hi @sgomes, thanks for making the changes! `afterLayoutFlush` is a framework primitive that many other components build on, and I believe it's worth spending some time making it really good. Like GC-ing the last args as soon as possible, preserving `this`...

One last thing: I suspect that the `this` preservation will not work because `scheduleRAF` returns an arrow function. Arrow function have `this` bound at creation time and it can never be rebound. We might need to change it to:
```js
const scheduleRAF = rafFunc => function( ...args ) {
  ...
};
```

Could you write a test that tests that `afterLayoutFlush` does all the scheduling and args passing correctly? Jest and JSDOM hopefully have the tools we need to mock animation frames and timeouts.

Reply:
My ad-hoc in-browser testing went fine (`this` included), but it's definitely better to have some unit tests for this. I'll get to work on them 👍

Reply:
... and it turns out my add-hoc testing went fine because I was testing the wrong thing. And once again, you were right, @jsnajdr 😄The arrow functions were binding `this` and stopping this `this` preservation from working correctly.

This is why it's definitely worth spending the time and write tests! Particularly for such a core library, as you mentioned.

This is looking better and better! 😁

Reply:
Actually, let me just add some tests for `.cancel()`, too.

Reply:
Tests added. Ready for another review :)